File: Sources/AdbShell/Services/IpRouteParser.swift
import Foundation

/// Достаёт IP-адрес устройства из вывода `adb shell ip route` — нужен, чтобы
/// после включения Wi-Fi отладки (`adb tcpip`) показать, куда подключаться
/// (`adb connect <ip>:<port>`), не заставляя лезть в Настройки → О телефоне.
enum IpRouteParser {
    static func parseDeviceIP(from output: String) -> String? {
        let lines = output.split(separator: "\n").map(String.init)
        // Предпочитаем Wi-Fi интерфейс (wlan*) — большинство прошивок его и
        // отдают первой строкой, но не полагаемся на порядок.
        let wlanLine = lines.first { $0.contains("wlan") && $0.contains(" src ") }
        let anyLine = lines.first { $0.contains(" src ") }
        guard let line = wlanLine ?? anyLine else { return nil }
        let parts = line.split(separator: " ").map(String.init)
        guard let srcIndex = parts.firstIndex(of: "src"), srcIndex + 1 < parts.count else { return nil }
        return parts[srcIndex + 1]
    }
}

File: src/main/adb/parsers/IpRouteParser.ts
// Порт Sources/AdbShell/Services/IpRouteParser.swift — достаёт IP устройства
// из вывода `adb shell ip route`, нужен после `adb tcpip` для подсказки
// `adb connect <ip>:<port>`.

export function parseDeviceIP(output: string): string | undefined {
  const lines = output.split('\n');
  const wlanLine = lines.find((l) => l.includes('wlan') && l.includes(' src '));
  const anyLine = lines.find((l) => l.includes(' src '));
  const line = wlanLine ?? anyLine;
  if (!line) return undefined;

  const parts = line.split(' ').filter(Boolean);
  const srcIndex = parts.indexOf('src');
  if (srcIndex === -1 || srcIndex + 1 >= parts.length) return undefined;
  return parts[srcIndex + 1];
}
